### PR TITLE
Mainline NPC traits mod

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary

```SUMMARY: Content "Mainline NPC traits mod"```

#### Purpose of change

NPC Traits mod was clashing with NPC backstories. Really, it should be mainline anyway as NPCs should have traits.

NOTE: I appear to have linted the entire classes.json during this. Apologies for the large change for reviewers, but I suppose this should be done eventually anyway.

#### Describe the solution

Mainlines NPC traits, giving all NPCs random trait groups in addition to background story traits.

This has some very cursory rebalancing of the mod, increasing the likelihood of several trait categories to encourage commonly taken traits from appearing in NPCs, and reducing some that aren't common in the general population such as albinism and schizophrenia.

In addition, it disables several food/thirst related traits as currently these features aren't well implemented in NPCs and are turned off by the 'disable NPC needs' mod to begin with. When NPC behaviour is better fleshed out, this can be revisited, although I'm not sure dealing with increased hunger and thirst in NPCs is much fun for anyone.

#### Describe alternatives you've considered

adding background stories to the NPC traits mod would also work, but requires maintenance of the mod.

#### Additional context

- [x] Add NPC random NPC starting traits to the list of class traits
- [x] Add NPC trait groups as their own file in the NPC folder
- [x] Rebalance NPC trait groups a bit to suit the meta
- [x] Remove NPC traits from mod folder
- [x] Lint classes.json and starting_traits.json
- [x] Review balance of traits and consider further rebalancing to create interesting and realistic NPCs
- [x] Test in-game

### CODE REVIEW

**Comment by Vasyan2006** [comment]
Don't you mind to increase probability of most traits from 10 to 20? In this case NPCs will have more traits (12 on average) and will be more diverse and interesting.

**Comment by I-am-Erk** [comment]
> Don't you mind to increase probability of most traits from 10 to 20? In this case NPCs will have more traits (12 on average) and will be more diverse and interesting.

I don't think NPCs need to be quite as min-maxed as players are. An NPC with that many traits is still possible, but should be fairly rare.

I did mostly increase trait frequency where I changed it, so more traits should be showing up in this version.

**Comment by Vasyan2006** [comment]
Trait groups have 50% to give a negative trait and having more random traits will not make a character stronger. I dont see how it can be min-maxed if very unlucky NPC will have twice more negative traits and none positive.

Martial arts traits are purely cosmetic, they will not give any combat style. Remove them until NPCs combat will be updated?

**Comment by I-am-Erk** [comment]
> Trait groups have 50% to give a negative trait and having more random traits will not make a character stronger. I dont see how it can be min-maxed if very unlucky NPC will have twice more negative traits and none positive.
> 
> Martial arts traits are purely cosmetic, they will not give any combat style. Remove them until NPCs combat will be updated?

I don't really mean that NPC's will be stronger, just that players having 12-20 traits normally doesn't mean NPCs need to have 12-20 traits normally.

I could remove martial arts, but like Liar and Truthteller, although they don't do anything themselves I could see them utilized later to do other things (eg. making it possible for NPCs to teach martial arts).

File: data/mods/NPC_Traits/modinfo.json [hunk 1]
```diff
@@ -1,11 +0,0 @@
-[
-  {
-    "type": "MOD_INFO",
-    "ident": "npc_traits",
-    "name": "NPC traits",
-    "authors": [ "Spencer Michaels" ],
-    "description": "Add an NPC class that spawns with starting traits, like the player does.",
-    "category": "misc_additions",
-    "dependencies": [ "dda" ]
```
Comment (cainiaowu) [inline data/mods/NPC_Traits/modinfo.json]: This needs to be marked as obsoleted instead of removing , don't break saves.

Comment (cainiaowu) [reply data/mods/NPC_Traits/modinfo.json]: you add one line `"obsolete": true` in the modinfo.json.

See https://github.com/CleverRaven/Cataclysm-DDA/pull/19398/commits/b93567b9093cda30f2dbbad2ecca88289ac67cc0

Comment (I-am-Erk) [reply data/mods/NPC_Traits/modinfo.json]: Thanks!

**Comment by I-am-Erk** [comment]
I'm not about to make alternate NPC versions of every starting trait to make them unpurifiable. If there's a flag I can add when assigning the trait I can add that.

**Comment by Vasyan2006** [comment]
>if NPC gets trait which should be possible to get only at character creation, and therefore should be unpurifable
Every NPCs trait is saved as a mutations and can be purifier. After few decialiters of purifier NPC will have zero traits. I suppose it is a bug and it mus t be fixed.

File: data/json/npcs/classes.json [hunk 2]
```diff
@@ -1,443 +1,314 @@
...
+      { "skill": "mechanics", "bonus": { "one_in": 2 } },
+      { "skill": "electronics", "bonus": { "rng": [ 0, 2 ] } },
+      { "skill": "speech", "bonus": { "rng": [ 1, 3 ] } },
+      { "skill": "barter", "bonus": { "rng": [ 3, 5 ] } }
+    ]
+  },
+  {
+    "type": "npc_class",
+    "id": "NC_SHOPKEEP",
+    "name": "Shopkeep",
```
Comment (ZhilkinSerg) [inline data/json/npcs/classes.json]:
```suggestion
     "name": "Shopkeep",
     "job_description" : "I'm a local shopkeeper.",
```

image

**Comment by DracoGriffin** [comment]
Will add to #24699 upon merge.

**Comment by ZhilkinSerg** [comment]
Removable traits could be a bug, but that should be addressed in another PR.

Duplicating traits is a no-go imo.

**Pull request merged**